imagechecker_postprocessing.rb

```diff
@@ -123,54 +175,81 @@ def writeTypeErrors(arr, file)
         output.puts r
       end
     end
+  else
+    logstring = 'n-a'
   end
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+## wrapping a Mcmlln::Tools method in a new method for this script; to return a result for json_logfile
+def overwriteFile(path, filecontents, logkey='')
+	Mcmlln::Tools.overwriteFile(path, filecontents)
+rescue => logstring
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
 end
 
+
 # ---------------------- PROCESSES
 
-filecontents = File.read(Bkmkr::Paths.outputtmp_html)
+filecontents = readOutputHtml('read_outputtmp-html')
 
-images = Mcmlln::Tools.dirListFiles(Bkmkr::Paths.project_tmp_dir_img)
+images = getFilesinDir(Bkmkr::Paths.project_tmp_dir_img, 'get_files_in_tmpImgDir')
 
-finalimages = Mcmlln::Tools.dirList(final_dir_images)
+finalimages = getFilesinDir(final_dir_images, 'get_files_in_finalImgDir')
 
-checkErrorFile(image_error)
+checkErrorFile(image_error, 'delete_image_error_file')
 
-# run method: listImages
-imgarr = listImages(filecontents)
+# run method: list unique images referenced in html
+imgarr = listImages(filecontents, 'list_images')
 
 # run method: checkImages
-format, supported, missing = checkImages(imgarr, images, finalimages, Bkmkr::Paths.project_tmp_dir_img)
+format, supported, missing = checkImages(imgarr, images, finalimages, Bkmkr::Paths.project_tmp_dir_img, 'check_images')
 
-# print a list of any unsupported image types
+# print a list of any unsupported image types to std log & json log
 unless format.nil? or format.empty? or !format
   puts "UNSUPPORTED IMAGE TYPES:"
   puts format
+  @log_hash['unsupported_image_types'] = format
 end
 
 # run method: convertImages
-corrupt, converted = convertImages(supported, Bkmkr::Paths.project_tmp_dir_img)
+corrupt, converted = convertImages(supported, Bkmkr::Paths.project_tmp_dir_img, 'convert_images')
+
+# print a list of converted and or corrupt images to json_logfile
+unless corrupt.nil? or corrupt.empty? or !corrupt
+  @log_hash['corrupt_images'] = corrupt
+end
+unless converted.nil? or converted.empty? or !converted
+  @log_hash['converted_images'] = converted
+end
```

Comment:
My inclination would be to just let the log write a nil value for these two keys, instead of omitting it from the log if there are no corrupt or converted images. But if there's a problem with using nil values, then this is ok.